flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/common/schema/factories/AvroSchemaFactory.java

```diff
@@ -45,4 +48,14 @@ public SchemaType type() {
 
         return AvroSchema.of(definition);
     }
+
+    @Override
+    public TypeInformation<T> createTypeInfo(SchemaInfo info) {
+        try {
+            Class<T> decodeClassInfo = decodeClassInfo(info);
+            return AvroUtils.getAvroUtils().createAvroTypeInfo(decodeClassInfo);
```

Comment:
If user didn't load the AVRO jar, we fall back to use the schema specific serializer.